Add api/metadata endpoints and support Metadata search (#1284)
web-platform-tests/wpt.fyi: api/query/README.md

      "product": "chrome-69",
      "status": "ok",
    }

#### link

> `link` query objecsts performs a search on a test that has a Metadata Link

Reviewer: ```suggestion
`link` query atoms perform a search for tests that have some matching link metadata
```

Reviewer: (And drop L146)
